Mu2eBTrk/inc/ParticleInfo.hh

```diff
@@ -41,9 +42,8 @@ namespace mu2e {
     //the following has to be mutable because BTrk holds this
     //with a const pointer
 
-    // Handle to the full particle data table.
     // Guaranteed valid throughout the job.
-    mutable GlobalConstantsHandle<ParticleDataList> pdt_;
+    ParticleDataList const& pdt_;
```

Comment:
presumably this is the same object as before.  Who holds that handle and makes sure the object stays in scope throughout the job?  Is that guaranteed by the service?

Reply:
> presumably this is the same object as before. Who holds that handle and makes sure the object stays in scope throughout the job? Is that guaranteed by the service?

Yes.  The ParticleDataList is created by GlobalConstantsService upon the first request.  It is created on the heap and stays in the same place for the duration of the job.  The object is owned by the GlobalConstantsService  and
it's destructor is called during the destructor of the GlobalConstantsService.

When you create a GlobalConstantsHandle<T> it gets an art::ServiceHandle<GlobalConstantsService> and  asks the service for a const pointer to the  object.
 
The above relies on the rules for what we allow into the GlobalConstantsService. 

Does that answer your question?